qiskit_ibm_runtime/fake_runtime_job.py

```diff
@@ -0,0 +1,80 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Qiskit fake runtime job."""
+
+from typing import Optional, Dict, Any
+
+from qiskit.primitives.primitive_job import PrimitiveJob
+from qiskit.providers import JobStatus, JobV1
+from qiskit.providers.fake_provider import FakeBackendV2 as FakeBackend
+
+# pylint: disable=cyclic-import
+# from .qiskit_runtime_service import QiskitRuntimeService
+
+
+class FakeRuntimeJob(JobV1):
+    """Representation of a runtime program execution on a simulator."""
+
+    def __init__(
+        self,
+        primitive_job: PrimitiveJob,
+        backend: FakeBackend,
+        job_id: str,
+        program_id: str,
+        # service: "qiskit_runtime_service.QiskitRuntimeService",
+        params: Optional[Dict] = None,
+        # creation_date: Optional[str] = None,
+        # user_callback: Optional[Callable] = None,
+        # result_decoder: Optional[Union[Type[ResultDecoder], Sequence[Type[ResultDecoder]]]] = None,
+        # tags: Optional[List] = None,
+    ) -> None:
+        """FakeRuntimeJob constructor."""
+        super().__init__(backend=backend, job_id=job_id)
+        # self._service = service  # do we need this?
+        self._primitive_job = primitive_job
+        self._job_id = job_id
+        self._params = params or {}
+        self._program_id = program_id
+
+    def result(self) -> Any:
+        """Return the results of the job."""
+        return self._primitive_job.result()
+
+    def cancel(self) -> None:
+        self._primitive_job.cancel()
+
+    def status(self) -> JobStatus:
+        return self._primitive_job.status()
```

Comment:
Why not just inherit `PrimitiveJob`, then you only need to add methods not already supported.

Reply:
I tried, but couldn't manage to do that, because I am receiving `primitive_job` as an object and couldn't find a way to construct the `FakeRuntimeObject` with a pre-constructed `super()` object. I would need something like a copy constructor.
I could simply use `PrimitiveJob` directly, but then I wouldn't be able to add any fields/methods.
Can you advise how to do this with inheritance?